lib/KBaseReport/utils/validation_utils.py

```diff
@@ -172,14 +311,52 @@ def _format_errors(errors, params):
     }
 }
 
+# Template + template data validation
+template_schema = {
+    'template_file': {
+        'type': 'string',
+        'minlength': 3,
+        'required': True,
+    },
+    'template_data_json': {
+        'type': 'string',
+        'validator': valid_json,
+    },
+}
+
 # Type validation for the extended report's File (see KIDL spec)
 extended_file_schema = {
     'type': 'dict',
     'schema': {
-        'name': {'type': 'string', 'nullable': True},
-        'shock_id': {'type': 'string', 'nullable': True},
-        'path': {'type': 'string', 'nullable': True},
-        'description': {'type': 'string', 'nullable': True},
-        'label': {'type': 'string', 'nullable': True}
+        'name': {
+            'type': 'string',
+            'required': True,
+            'minlength': 1,
+        },
+        'shock_id': {
+            'type': 'string',
+            'excludes': ['path', 'template'],
+            'required': True,
+        },
+        'path': {
+            'type': 'string',
+            'excludes': ['shock_id', 'template'],
+            'required': True,
+            'validator': valid_file_or_dir,
+        },
+        'template': {
+            'type': 'dict',
+            'excludes': ['path', 'shock_id'],
+            'required': True,
+            'schema': template_schema,
+        },
+        'description': {
+            'type': 'string',
+            'nullable': True,
+        },
+        'label': {
+            'type': 'string',
+            'nullable': True,
+        },
```

Comment:
Items marked as `required` and with `excludes` params acts like an exclusive OR.